packages/rn-tester/android/app/src/main/java/com/facebook/react/uiapp/component/MyNativeView.kt

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+package com.facebook.react.uiapp.component
+
+import android.content.Context
+import android.graphics.Color
+import android.graphics.drawable.GradientDrawable
+import android.view.View
+import com.facebook.react.bridge.Arguments
+import com.facebook.react.bridge.ReactContext
+import com.facebook.react.bridge.WritableArray
+import com.facebook.react.bridge.WritableMap
+import com.facebook.react.uimanager.UIManagerHelper
+import com.facebook.react.uimanager.events.Event
+import com.facebook.react.uimanager.events.EventDispatcher
+import com.facebook.react.uimanager.events.RCTEventEmitter
+
+class MyNativeView(context: Context) : View(context) {
+  private var currentColor = 0
+  private var background: GradientDrawable = GradientDrawable()
+
+  override fun setBackgroundColor(color: Int) {
+    if (color != currentColor) {
+      background.setColor(color)
+      currentColor = color
+      emitNativeEvent(color)
+      setBackground(background)
+    }
+  }
+
+  fun setCornerRadius(cornerRadius: Float) {
+    background.setCornerRadius(cornerRadius)
+    setBackground(background)
+  }
+
+  private fun emitNativeEvent(color: Int) {
+    val event = Arguments.createMap()
+    val backgroundColor = Arguments.createMap()
+    val hsv = FloatArray(3)
+
+    Color.colorToHSV(color, hsv)
+    backgroundColor.putDouble("hue", hsv[0].toDouble())
+    backgroundColor.putDouble("saturation", hsv[1].toDouble())
+    backgroundColor.putDouble("brightness", hsv[2].toDouble())
+    backgroundColor.putDouble("alpha", Color.alpha(color).toDouble())
```

Comment:
```suggestion
    val backgroundColor = Arguments.createMap().apply {
      putDouble("hue", hsv[0].toDouble())
      putDouble("saturation", hsv[1].toDouble())
      putDouble("brightness", hsv[2].toDouble())
      putDouble("alpha", Color.alpha(color).toDouble())
    }
```

Reply:
Thank you for your feedback. Code updated